library/CM/Migration/Loader.php

```
@@ -0,0 +1,54 @@
+<?php
+
+class CM_Migration_Loader implements CM_Service_ManagerAwareInterface {
+
+    const MIGRATION_DIR = 'migration';
+
+    use CM_Service_ManagerAwareTrait;
+
+    /**
+     * @param CM_Service_Manager $serviceManager
+     */
+    public function __construct(CM_Service_Manager $serviceManager) {
+        $this->setServiceManager($serviceManager);
+    }
+
+    /**
+     * @param string $name
+     * @return CM_Migration_Script|null
+     */
+    public function findScript($name) {
+        return \Functional\first($this->getScriptList(), function (CM_Migration_Script $script) use ($name) {
```

Theoretically there can be multiple scripts with the same name, but in different namespaces. Such a scenario is problematic, and maybe we should throw an exception in that case?